test/uk/gov/hmrc/cgtpropertydisposals/service/returns/ReturnsServiceSpec.scala

```diff
@@ -1460,6 +1460,281 @@ class ReturnsServiceSpec extends WordSpec with Matchers with MockFactory {
+            Some(
+              ReturnCharge(
+                chargeReference,
+                AmountInPence(1100L),
+                LocalDate.of(2020, 3, 11)
+              )
+            ),
+            None
+          )
+          val (fromDate, toDate)   = (taxYear.startDateInclusive, taxYear.endDateExclusive)
+          val responseJsonBody     =
+            Json.parse(s"""
+                          |{
+                          |"processingDate":"2020-02-20T09:30:47Z",
+                          |"ppdReturnResponseDetails": {
+                          |     "chargeType": "Late Penalty",
+                          |     "chargeReference":"$chargeReference",
+                          |     "amount":11.0,
+                          |     "dueDate":"2020-03-11",
+                          |     "formBundleNumber":"$formBundleId",
+                          |     "cgtReferenceNumber":"${submitReturnRequest.subscribedDetails.cgtReference}"
+                          |  }
+                          |}
+                          |""".stripMargin)
+
+          inSequence {
+            mockAuditSubmitReturnEvent(
+              submitReturnRequest.subscribedDetails.cgtReference,
+              desSubmitReturnRequest,
+              submitReturnRequest.agentReferenceNumber
+            )
+            mockSubmitReturn(
+              submitReturnRequest.subscribedDetails.cgtReference,
+              desSubmitReturnRequest
+            )(Right(HttpResponse(200, responseJsonBody, Map.empty[String, Seq[String]])))
+            mockAuditSubmitReturnResponseEvent(
+              200,
+              Some(responseJsonBody),
+              desSubmitReturnRequest,
+              submitReturnRequest.subscribedDetails.name,
+              submitReturnRequest.agentReferenceNumber
+            )
+            mockSaveAmendReturnList(submitReturnRequest)(Right(()))
+            mockGetFinancialData(submitReturnRequest.subscribedDetails.cgtReference, fromDate, toDate)(
+              Right(HttpResponse(200, desFinancialDataResponse, Map.empty[String, Seq[String]]))
+            )
+            mockSendReturnSubmitConfirmationEmail(submitReturnResponse, submitReturnRequest.subscribedDetails)(
+              Right(HttpResponse(ACCEPTED, emptyJsonBody))
+            )
+            mockAuditReturnConfirmationEmailEvent(
+              submitReturnRequest.subscribedDetails.emailAddress.value,
+              submitReturnRequest.subscribedDetails.cgtReference.value,
+              formBundleId
+            )
+          }
+
+          await(returnsService.submitReturn(submitReturnRequest, None).value) shouldBe Right(submitReturnResponse)
+
+        }
+
+        "there are two DesFinancialTransactions which matches return's charge reference" in {
+
+          val formBundleId    = "804123737752"
+          val chargeReference = "XCRG9448959757"
+
+          val submitReturnRequest    = sample[SubmitReturnRequest].copy(
+            originalReturnFormBundleId = None
```

Comment:
shouldn't this be `Some(..)`? delta charges should only apply when it's an amend return

Reply:
how does this test pass if it's not an amend return and you're expecting a `DeltaCharge` at the end of it?